plugins/out_loki/loki.c

```diff
@@ -1205,6 +1205,11 @@ static void cb_loki_flush(const void *data, size_t bytes,
         flb_http_basic_auth(c, ctx->http_user, ctx->http_passwd);
     }
 
+    /* Bearer token auth header */
```

Comment:
I followed Loki's own ruby implementation for Fluentd found at https://github.com/grafana/loki/blob/main/clients/cmd/fluentd/lib/fluent/plugin/out_loki.rb#L215. Their implementation shows that if both have data they will be added but now looking more into how ruby implements the `basic_auth` call that happens at https://github.com/ruby/ruby/blob/d92f09a5eea009fa28cd046e9d0eb698e3d94c5c/lib/net/http/header.rb#L488, it will overwrite instead of add like happens in this plugin. I'll make the behavior more similar with basic http auth winning.